plugin/src/main/java/com/iridium/iridiumskyblock/commands/ShopCommand.java

```diff
@@ -0,0 +1,79 @@
+package com.iridium.iridiumskyblock.commands;
+
+import com.iridium.iridiumskyblock.IridiumSkyblock;
+import com.iridium.iridiumskyblock.api.IridiumSkyblockAPI;
+import com.iridium.iridiumskyblock.database.User;
+import com.iridium.iridiumskyblock.gui.ShopCategoryGUI;
+import com.iridium.iridiumskyblock.gui.ShopOverviewGUI;
+import com.iridium.iridiumskyblock.shop.ShopCategory;
+import com.iridium.iridiumskyblock.utils.StringUtils;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+
+/**
+ * Command which opens the island shop.
+ */
+public class ShopCommand extends Command {
+
+    /**
+     * The default constructor.
+     */
+    public ShopCommand() {
+        super(Collections.singletonList("shop"), "Opens the Island shop", "", true);
+    }
+
+    /**
+     * Executes the command for the specified {@link CommandSender} with the provided arguments. Not
+     * called when the command execution was invalid (no permission, no player or command
+     * disabled).
+     *
+     * @param sender The CommandSender which executes this command
+     * @param arguments The arguments used with this command. They contain the sub-command
+     */
+    @Override
+    public void execute(CommandSender sender, String[] arguments) {
+        Player player = (Player) sender;
+        User user = IridiumSkyblockAPI.getInstance().getUser(player);
+        if (!user.getIsland().isPresent()) {
+            player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().dontHaveIsland.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
```

Comment:
Do we need an island to open island shop? shouldnt impact us buying/selling stuff